### PR TITLE
v3 - support image assets from shared Sanity project

### PR DESCRIPTION
**Pull request by mathiazom**
See #699 

Added `useNextSanityGlobalImage` to search both studio and shared when rendering images. Also refactored the use of image hooks and rendering.

Before | **After**
--- | ---
image | image

Storybook should not be affected, but a fix was added to add the required width and height props to Next.js Image:

image

## Checklist

Please ensure that you’ve completed the following checkpoints before submitting your pull request:

- [x] **Documentation**: Relevant documentation has been added or updated (if applicable).
- [x] **Testing**: Have you tested your changes thoroughly?

### CODE REVIEW

**Review by anemne — APPROVED** [review]

**Comment by christinaroise** [comment]
could you add screenshots of how this affects stories?

**Comment by mathiazom** [comment]
> could you add screenshots of how this affects stories?

Should not affect stories, but they were actually broken already... Added a fix now.

File: src/components/image/SanityImage.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  );
+};
+
+export function SanityImage({ image }: { image: IImage }) {
+  if (image?.src) {
+    return (
+      <Image
+        alt={image?.alt || ""}
+        src={image.src.src}
+        style={{ objectFit: "cover", height: "100%" }}
```
Comment (christinaroise) [inline src/components/image/SanityImage.tsx:71]: Discovered that this fix might be needed for a few cases

```suggestion
        style={{ objectFit: "cover", height: "100%", width: "100%"  }}
```

Comment (christinaroise) [reply src/components/image/SanityImage.tsx:71]: could then be that you don't need the width and height props below, but stories might want them anyway.. 🤷

**Review by christinaroise — APPROVED** [review]

File: src/components/image/SanityImage.tsx (hunk 1 above)
Comment (mathiazom) [reply src/components/image/SanityImage.tsx:71]: `width` and `height` are required props for [Next.js Image](https://nextjs.org/docs/app/api-reference/components/image), but we can add `width: "100%"` if it's needed :+1:

**Pull request merged**